### ISSUE TITLE
Support `safe` mode

### ISSUE THREAD

**Issue by moteus** [daurnimator/mmdblua#9]
I prefer if function returns error instead raise Lua errors.
E.g. if i get info from untrasted source I want return either info or error message.
Now I have to use `pcall` for such thing.
I like way how it was implemented in `cjson` library.
It just provide 2 module `cjson` and `cjson.safe`.

**Comment by daurnimator** [comment]
I agree. Happy to add functions that return nil, err, errno and keep old constructor for compatibility

**Comment by daurnimator** [comment]
Let me know what you think of #10. Pick me some better function names if you can :)
